enhancements/installer/component-selection.md

```diff
@@ -0,0 +1,381 @@
+  one based on the chosen install configuration (HA vs single node) instead of having components read/reconcile
+  the infrastructure resource.
+
+Current plan is to constrain this functionality to capability level controls.
+
+2. How does the admin enable a component post-install if they change their mind about what components
+they want enabled?  Do we need/want to allow this?
+
+Turning on a component later is relatively easy (we expose a config resource for the CVO that defines
+the filter, we allow the user to remove items from the filter, the CVO will apply the previously
+filtered resources during the next reconciliation).
+
+Turning off a component later is more problematic because
+a) The CVO doesn't delete resources today, so that would be a new thing to teach it to do.
+b) Just deleting the resources for the component isn't sufficient, as the component also needs to clean
+itself up in case it created any additional resources on the cluster or contributed any configuration.
+
+Therefore we plan to support turning a component on, but not turning it off.
+
+3. What are the implications for upgrades if a future upgrade would add a component or resource which would
+have been filtered out during install time?  
+
+There should be no implication here, the CVO has the list of annotations it will filter based on, if the
+new resources match those annotations, the new resources will also be filtered(never applied to the cluster).
+
+4. How prescriptive do we want to be about what can/can't be turned off?  Components need to opt into
+this by annotating their resources, so it's not completely arbitrary.
+
+This will need to be evaluated on a case by case basis as a component considers adding the annotation
+to its resources that will allow it to be filtered out/disabled.
+
+5. What if a user specifies a capability name to exclude that doesn't match anything?
+
+The installer won't know it's not going to match anything, so it should do nothing.  The CVO
+can tell it doesn't match anything, so it can provide a warning on the clusterversion resource, but
+it should not be an error since a user might put a keyword into the list in anticipation of upgrading
+to a new OCP version where the keyword will match (and thus exclude) a new capability, or perhaps
+the keyword used to match a capability that no longer exists after an upgrade.
+
+Along the same lines, if a user both "Excludes" and "Includes" a capability(if we ever add Include as
+a thing you can do), that should be treated as an error/invalid configuration.
+
+### Open Question
+
+1. What to do for components where disabling them has implications on other components or the way certain
+apis behave.  Example: disabling the internal registry changes the behavior of imagestreams
+(can't push to the imagestream anymore to push content to the internal registry) as well as the assumptions
+made by tools like new-app/new-build (create imagestreams that push to the internal registry).
```

Comment:
How do these components work today with clusters that have [`managementStat: Removed` registries][1]?

[1]: https://docs.openshift.com/container-platform/4.8/registry/configuring-registry-operator.html#registry-operator-configuration-resource-overview_configuring-registry-operator